src/simsopt/geo/magneticfield.py

```diff
@@ -1,139 +1,133 @@
-        if self._B is None:
-            self.compute(self.points, compute_derivatives)
-        return self._B
-
-    def dB_by_dX(self, compute_derivatives=1):
-        if self._dB_by_dX is None:
-            assert compute_derivatives >= 1
-            self.compute(self.points, compute_derivatives)
-        return self._dB_by_dX
-
-    def d2B_by_dXdX(self, compute_derivatives=2):
-        if self._d2B_by_dXdX is None:
-            assert compute_derivatives >= 2
-            self.compute(self.points, compute_derivatives)
-        return self._d2B_by_dXdX
-
-    def A(self, compute_derivatives=0):
-        if self._A is None:
-            assert compute_derivatives >= 0
-            self.compute_A(self.points, compute_derivatives)
-        return self._A
-
-    def dA_by_dX(self, compute_derivatives=1):
-        if self._dA_by_dX is None:
-            assert compute_derivatives >= 1
-            self.compute_A(self.points, compute_derivatives)
-        return self._dA_by_dX
-
-    def d2A_by_dXdX(self, compute_derivatives=2):
-        if self._d2A_by_dXdX is None:
-            assert compute_derivatives >= 2
-            self.compute_A(self.points, compute_derivatives)
-        return self._d2A_by_dXdX
-
-    def compute(self, points, compute_derivatives=0):
-        raise NotImplementedError('Computation of B field not implemented. Needs to be overwritten by childclass.')
-
-    def compute_A(self, points, compute_derivatives=0):
-        raise NotImplementedError('Computation of potential A not implemented. Needs to be overwritten by childclass.')
+        """Clear the cache."""
+        sopp.MagneticField.invalidate_cache(self)
 
     def __add__(self, other):
+        """Add two magnetic fields."""
         return MagneticFieldSum([self, other])
 
     def __mul__(self, other):
+        """Multiply a field with a scalar."""
         return MagneticFieldMultiply(other, self)
 
     def __rmul__(self, other):
+        """Multiply a field with a scalar."""
         return MagneticFieldMultiply(other, self)
 
+    def to_vtk(self, filename, nr=10, nphi=10, nz=10, rmin=1.0, rmax=2.0, zmin=-0.5, zmax=0.5):
+        """Export the field evaluated on a regular grid for visualisation with e.g. Paraview."""
+        from pyevtk.hl import gridToVTK
+        rs = np.linspace(rmin, rmax, nr, endpoint=True)
+        phis = np.linspace(0, 2*np.pi, nphi, endpoint=True)
+        zs = np.linspace(zmin, zmax, nz, endpoint=True)
+
+        R, Phi, Z = np.meshgrid(rs, phis, zs)
+        X = R * np.cos(Phi)
+        Y = R * np.sin(Phi)
+        Z = Z
+
+        RPhiZ = np.zeros((R.size, 3))
+        RPhiZ[:, 0] = R.flatten()
+        RPhiZ[:, 1] = Phi.flatten()
+        RPhiZ[:, 2] = Z.flatten()
+
+        self.set_points_cyl(RPhiZ)
+        vals = self.B()
+        contig = np.ascontiguousarray
+        gridToVTK(filename, X, Y, Z, pointData={"B": (contig(vals[:, 0]), contig(vals[:, 1]), contig(vals[:, 2]))})
+
 
 class MagneticFieldMultiply(MagneticField):
```

Comment:
Do we need a class to do this? Can't we use use numpy multiply operation? Another approach could be to write a class method.